src/overlays/gamestates/ovl_file_choose/z_file_copy_erase.c

```diff
@@ -0,0 +1,1253 @@
+        ABS_ALT(this->buttonYOffsets[this->buttonIndex] - sChooseFileYOffsets[this->buttonIndex]) / this->actionTimer;
+
+    if (this->buttonYOffsets[this->buttonIndex] >= sChooseFileYOffsets[this->buttonIndex]) {
+        this->buttonYOffsets[this->buttonIndex] -= yStep;
+    } else {
+        this->buttonYOffsets[this->buttonIndex] += yStep;
+    }
+
+    for (i = 0; i < 3; i++) {
+        if (i != this->buttonIndex) {
+            this->fileButtonAlpha[i] += 200 / 4;
+
+            if (!gSaveContext.flashSaveAvailable) {
+                if (NO_FLASH_SLOT_OCCUPIED(sramCtx, i)) {
+                    this->nameBoxAlpha[i] = this->nameAlpha[i] = this->fileButtonAlpha[i];
+                    this->connectorAlpha[i] += 255 / 4;
+                }
+            } else if (SLOT_OCCUPIED(this, i)) {
+                this->nameBoxAlpha[i] = this->nameAlpha[i] = this->fileButtonAlpha[i];
+                this->connectorAlpha[i] += 255 / 4;
+            }
+        }
+    }
+
+    this->titleAlpha[FS_TITLE_CUR] -= 255 / 4;
+    this->titleAlpha[FS_TITLE_NEXT] += 255 / 4;
+    this->actionTimer--;
+
+    if (this->actionTimer == 0) {
+        this->buttonYOffsets[this->buttonIndex] = sChooseFileYOffsets[this->buttonIndex];
+        this->actionTimer = 4;
+        this->buttonIndex = FS_BTN_ERASE_QUIT;
+        this->titleLabel = this->nextTitleLabel;
+        this->titleAlpha[FS_TITLE_CUR] = 255;
+        this->titleAlpha[FS_TITLE_NEXT] = 0;
+        this->configMode = CM_ERASE_SELECT;
+    }
+}
+
+/**
+ * Wait for an initial delay, then start fading out the selected file.
+ * The actual file deletion occurs in this function
+ * Update function for `CM_ERASE_ANIM_1`
+ */
+void FileSelect_EraseAnim1(GameState* thisx) {
+    static s16 D_80814E80;
+    FileSelectState* this = (FileSelectState*)thisx;
+    SramContext* sramCtx = &this->sramCtx;
+
+    if (sEraseDelayTimer == 0) {
+        if (this->actionTimer == 4) {
+            D_80814E80 = 1;
+        }
+
+        if (this->actionTimer != 0) {
+            this->titleAlpha[FS_TITLE_CUR] -= 255 / 4;
+            this->titleAlpha[FS_TITLE_NEXT] += 255 / 4;
+            this->fileInfoAlpha[this->selectedFileIndex] -= 200 / 4;
+            this->confirmButtonAlpha[FS_BTN_CONFIRM_YES] -= 200 / 4;
+            this->confirmButtonAlpha[FS_BTN_CONFIRM_QUIT] -= 200 / 4;
+        }
+
+        this->fileNamesY[this->selectedFileIndex] -= D_80814E80;
+        D_80814E80 += 2;
+
+        this->actionTimer--;
+        if (this->actionTimer == 0) {
+            this->configMode = CM_ERASE_ANIM_2;
+            this->titleLabel = this->nextTitleLabel;
+            this->titleAlpha[FS_TITLE_CUR] = 255;
+            this->titleAlpha[FS_TITLE_NEXT] = this->connectorAlpha[this->selectedFileIndex] = 0;
+
+            //! FAKE: there should be a better chained assignment
+            this->confirmButtonAlpha[FS_BTN_CONFIRM_YES] = this->confirmButtonAlpha[FS_BTN_CONFIRM_QUIT] = 0;
+            if (1) {}
+            if (0) {}
```

Comment:
1 and 0?

Reply:
Yup
https://decomp.me/scratch/UTHQu

Reply:
Can do 3 `if (1) {}`s instead, although I don't think it's really better (or worse).